lib/viewmodels/user_centauri_points_view_model.dart

```diff
@@ -0,0 +1,41 @@
+import "dart:async";
+
+import "package:hooks_riverpod/hooks_riverpod.dart";
+import "package:proxima/models/database/user/user_id_firestore.dart";
+import "package:proxima/services/database/user_repository_service.dart";
+
+/// A view model that provides the number of centauri points of a user
+/// given its user id. Allow a null [arg] (user id) to return null, which
+/// is useful for loading states.
+class UserCentauriPointsViewModel
+    extends FamilyAsyncNotifier<int?, UserIdFirestore?> {
+  UserCentauriPointsViewModel();
+
+  @override
+  Future<int?> build(UserIdFirestore? arg) async {
+    final userDataBase = ref.watch(userRepositoryServiceProvider);
+
+    if (arg == null) {
+      return null;
+    }
+    final userID = arg;
+
+    final user = await userDataBase.getUser(userID);
+    final userCentauri = user.data.centauriPoints;
+    return userCentauri;
+  }
+
+  /// Refresh the number of centauri points of the user.
+  Future<void> refresh() async {
+    state = await AsyncValue.guard(() => build(arg));
```

Comment:
If you decide to apply my previous comment, I'd recommend to also use the following here:
```suggestion
    state = const AsyncValue.loading();
    state = await AsyncValue.guard(() => build(arg));
```

This allows to let the UI decide whether it want to show the user the value is loading, or instead pick the last existing value.

Reply:
You are right ! 

Added in: [5154d96](https://github.com/ProximaEPFL/proxima/pull/348/commits/5154d966387b81f654a70a210edc41deec765ae3)

Thank you !